pkg/iac/terraform/state/enumerator/s3.go

```diff
@@ -14,20 +14,17 @@ import (
 	"github.com/cloudskiff/driftctl/pkg/iac/config"
 )
 
-type S3EnumeratorConfig struct {
-	Bucket *string
-	Prefix *string
-}
-
 type S3Enumerator struct {
 	config config.SupplierConfig
 	client s3iface.S3API
 }
 
-func NewS3Enumerator(config config.SupplierConfig) *S3Enumerator {
+func NewS3Enumerator(config config.SupplierConfig, envProxy *EnvProxy) *S3Enumerator {
+	envProxy.SetProxy("DCTL_S3_", "AWS_", false)
 	sess := session.Must(session.NewSessionWithOptions(session.Options{
 		SharedConfigState: session.SharedConfigEnable,
 	}))
```

Comment:
You can initialize EnvProxy directly in NewS3Enumerator since you'll rollback in the same function